src/index.js

```diff
@@ -160,6 +160,10 @@ class Router extends Component {
 		return props.url!==this.props.url || props.onChange!==this.props.onChange;
 	}
 
+	componentWillReceiveProps(props) {
+		if (props.url && props.url!==this.state.url) this.routeTo(props.url) && setUrl(props.url);
```

Comment:
Passing a new URL to `<Router>` should not call `setUrl()`, since doing so will change all other Routers on the page.